core/drivers/clk/sam/at91_programmable.c

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: GPL-2.0+ or BSD-3-Clause
+/*
+ *  Copyright (C) 2013 Boris BREZILLON <b.brezillon@overkiz.com>
+ *  Copyright (C) 2021 Microchip
+ */
+
+#include <io.h>
+#include <kernel/delay.h>
+#include <kernel/panic.h>
+#include <mm/core_memprot.h>
+#include <types_ext.h>
+
+#include "at91_clk.h"
+
+#define PROG_ID_MAX		7
+
+#define PROG_STATUS_MASK(id)	(1 << ((id) + 8))
+#define PROG_PRES(layout, pckr)	\
+({ \
+	typeof(layout) __layout = layout; \
+	(((pckr) >> (__layout)->pres_shift) & (__layout)->pres_mask); \
+})
```

Comment:
Indentation (+ should be an empty line between variable definitions and instructions)
```c
#define PROG_PRES(layout, pckr)	\
	({ \
		typeof(layout) __layout = layout; \
		\
		(((pckr) >> (__layout)->pres_shift) & (__layout)->pres_mask); \
	})